tests/test_nblist.py

```diff
@@ -143,60 +215,70 @@ def test_neighborlist():
             coords = coords[perm]
 
         ref_ixn_list = build_reference_ixn_list(coords, box, cutoff)
-
-        for nblist in (custom_ops.Neighborlist_f32(size), custom_ops.Neighborlist_f64(size)):
-
+        for nblist in nblists:
+            # Resize the nblist accordingly
+            nblist.resize(size)
             # Run twice to ensure deterministic results
             for _ in range(2):
                 test_ixn_list = nblist.get_nblist(coords, box, cutoff)
 
                 assert len(ref_ixn_list) == len(test_ixn_list)
 
-                for bidx, (a, b) in enumerate(zip(ref_ixn_list, test_ixn_list)):
-                    if sorted(a) != sorted(b):
-                        print("TESTING bidx", bidx)
-                        print(sorted(a))
-                        print(sorted(b))
-                    np.testing.assert_equal(sorted(a), sorted(b))
+                assert_ixn_lists_are_equal(ref_ixn_list, test_ixn_list)
 
 
-def test_neighborlist_ligand_host_invalid_parameters():
-    cols = 10
-    rows = 5
-    box = np.eye(1)
-    cutoff = 1.0
+def test_neighborlist_resize():
+    N = 3
+
+    # Verify that the sizes of the rows and columns match how the NBlist was constructed
+    for nblist in (
+        custom_ops.Neighborlist_f32(N),
+        custom_ops.Neighborlist_f64(N),
+    ):
+        with pytest.raises(RuntimeError) as e:
+            nblist.resize(0)
+        assert "size is must be at least 1" == str(e.value)
 
-    # Constructing NBlist with rows > cols is invalid
-    with pytest.raises(RuntimeError) as e:
-        custom_ops.Neighborlist_f32(rows, cols)
-    assert "NR is greater than NC" in str(e.value)
+        with pytest.raises(RuntimeError) as e:
+            nblist.resize(N * 3)
```

Comment:
nit: could make this stricter?
```suggestion
            nblist.resize(N + 1)
```